src/Microsoft.Health.Dicom.Core/Features/Indexing/DicomDatasetReindexValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+
+namespace Microsoft.Health.Dicom.Core.Features.Indexing
+{
+    public class DicomDatasetReindexValidator : IDicomDatasetReindexValidator
+    {
+        private readonly IDicomElementMinimumValidator _minimumValidator;
+
+        public DicomDatasetReindexValidator(IDicomElementMinimumValidator minimumValidator)
+        {
+            _minimumValidator = EnsureArg.IsNotNull(minimumValidator, nameof(minimumValidator));
+        }
+        public IReadOnlyCollection<QueryTag> Validate(DicomDataset dicomDataset, IReadOnlyCollection<QueryTag> queryTags)
```

Comment:
Left a comment on the interface, if possible we should avoid these multiple validator classes. Logically they are just indexing like STOW, you may want to just control some behaviors through params